pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -32,6 +32,46 @@ var otherToleration = corev1.Toleration{
 	Effect:   corev1.TaintEffectNoExecute,
 }
 
+func checkDeploymentHasEnvVar(t *testing.T, deployment *appsv1.Deployment, name string, expectValue bool, expectedValue string) {
+	t.Helper()
+
+	var (
+		actualValue string
+		foundValue  bool
+	)
+	for _, envVar := range deployment.Spec.Template.Spec.Containers[0].Env {
+		if envVar.Name == name {
+			foundValue = true
+			actualValue = envVar.Value
+			break
+		}
+	}
+	switch {
+	case !expectValue && foundValue:
+		t.Errorf("router Deployment has unexpected %s setting: %q", name, actualValue)
+	case expectValue && !foundValue:
+		t.Errorf("router Deployment is missing %v", name)
+	case expectValue && expectedValue != actualValue:
+		t.Errorf("router Deployment has unexpected %s setting: expected %q, got %q", name, expectedValue, actualValue)
+	}
+}
+
+func checkDeploymentHasContainer(t *testing.T, deployment *appsv1.Deployment, name string, expect bool) {
+	t.Helper()
+
+	for _, container := range deployment.Spec.Template.Spec.Containers {
+		if container.Name == name {
+			if expect {
+				return
+			}
+			t.Errorf("router Deployment has unexpected %q container", name)
+		}
+	}
+	if expect {
+		t.Errorf("router Deployment has no %q container", name)
+	}
+}
+
 func checkRollingUpdateParams(t *testing.T, deployment *appsv1.Deployment, maxUnavailable intstr.IntOrString, maxSurge intstr.IntOrString) {
```

Comment:
Nice little refactor here, thanks — one more thing it made me think of (not required, just a sidebar discussion) is you can now express your expectations with a struct and make a table test out of the env assertions, like

```
{
  env{name: WildcardRouteAdmissionPolicy, equals: "false"},
  env{name: "ROUTER_USE_PROXY_PROTOCOL", equals: nil},
}
```

And feed them into your assertion function. Likewise with container checks. And then keep adding sugar until we reinvent Ginkgo 😁